cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -115,6 +115,13 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, String funct
                 throw createLauncherException("you are trying to run a ballerina file inside a package within a " +
                                                       "project. Try running 'ballerina run <package-name>'");
             }
+            // Checks if the source path is a package
```

Comment:
This check shouldn't be done here in this class. This is a low-level check that should be done somewhere else.

We are calling the "ProjectDirs.containsSourceFiles" method here. It is a class that should only be known by the compiler not the build command.

Reply:
Moved the check to the compiler with ada9c2d